File: stuff/pyfemkit/coreClasses/baseFEClass.py
import numpy as np
from commonFunctions.checkIntegerMatrix import checkIntegerMatrix

class baseFEClass():
    def __init__(self):
        self.nodes = np.array([],dtype=float)
        self.edof = np.array([],dtype=int)             # local node based edof (element degrees of freedom table)
        self.globalFullEdof = np.array([],dtype=int)     # global dof based edof
        self.globalNodesDof = np.array([],dtype=int)    # global node based dof (degrees of freedom)
        self.dimension = 0
#set methods
    def setedof(self,input): #set. nicht möglich
        checkIntegerMatrix(input)
        self.edof=int(input)

    def setglobalFullEdof(self,input):
        checkIntegerMatrix(input)
        self.globalFullEdof=int(input)

    def setglobalNodesDof(self,input):
        checkIntegerMatrix(input)
        self.globalNodesDof = int(input)
        
    def setdimension(self,input):
        checkIntegerMatrix(input)
        if (input>3 and input <1):
            self.dimension = input

File: stuff/tag/2021_03_15/coreClasses/baseFEClass.m
classdef (Abstract) baseFEClass < handle
    properties
        nodes
        edof                        % local node based edof (element degrees of freedom table)
        globalFullEdof              % global dof based edof
        globalNodesDof              % global node based dof (degrees of freedom)
        dimension
    end
    methods
        %% set and get methods
        function set.edof(obj,input)
            checkIntegerMatrix(input);
            obj.edof = int32(input);
        end
        function set.globalFullEdof(obj,input)
            checkIntegerMatrix(input);
            obj.globalFullEdof = int32(input);
        end
        function set.globalNodesDof(obj,input)
            checkIntegerMatrix(input);
            obj.globalNodesDof = int32(input);
        end
        function set.dimension(obj,input)
            checkIntegerMatrix(input);
            assert((input <= 3) & (input >= 1));
            obj.dimension = input;
        end
    end
end